static/js/server_events_dispatch.js

```diff
@@ -313,17 +313,34 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
             // than requiring a reload or page resize.
         }
         if (event.setting_name === 'emojiset') {
-            var sprite = new Image();
-            sprite.onload = function () {
-                $("#emoji-spritesheet").attr('href', "/static/generated/emoji/" + page_params.emojiset + "_sprite.css");
+            if (page_params.emojiset === 'text') {
+                page_params.emoji_alt_code = true;
+                // Rerender the whole message list UI
+                home_msg_list.rerender();
+                if (current_msg_list === message_list.narrowed) {
+                    message_list.narrowed.rerender();
+                }
+                // destroy the spinner and display success message
                 if ($("#display-settings-status").length) {
                     loading.destroy_indicator($("#emojiset_spinner"));
                     $("#emojiset_select").val(page_params.emojiset);
                     ui_report.success(i18n.t("Emojiset changed successfully!!"),
-                                      $('#display-settings-status').expectOne());
+                                    $('#display-settings-status').expectOne());
                 }
-            };
-            sprite.src = "/static/generated/emoji/sheet_" + page_params.emojiset + "_32.png";
+            } else {
+                page_params.emoji_alt_code = false;
+                var sprite = new Image();
+                sprite.onload = function () {
+                    $("#emoji-spritesheet").attr('href', "/static/generated/emoji/" + page_params.emojiset + "_sprite.css");
+                    if ($("#display-settings-status").length) {
+                        loading.destroy_indicator($("#emojiset_spinner"));
```

Comment:
I think duplication of this piece of code for destroying spinner can be avoided?

Reply:
One time it is called when the emojiset has been loaded and other time for text. If I factor it out and write outside if-else then it won't be correct if the emojiset isn't loaded. What should I do?